addons/base_import/models/base_import.py

```diff
@@ -1607,7 +1607,11 @@ def _handle_multi_mapping(self, import_fields, input_file_data):
                 # merge data if necessary
                 if field_type in CONCAT_SEPARATOR_IMPORT:
                     separator = CONCAT_SEPARATOR_IMPORT[field_type]
-                    new_record.append(separator.join(record[idx] for idx in indexes if record[idx]))
+                    if field_type == 'char' and field.trim:
+                        # Trim trailing whitespaces before joining
+                        new_record.append(separator.join(record[idx].strip() for idx in indexes if record[idx]))
```

Comment:
not sure if `isinstance(record[id], str)`. But I think it would be better to drop `record[idx].strip() == ''`